solutions/silver/cf-1368D.mdx

```diff
@@ -0,0 +1,139 @@
+---
+id: cf-1368D
+source: CF
+title: AND, OR and square sum
+author: Paul Chen
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/79027)
+
+# Explanation
+
+**Observation 1**: Notice that $x \ AND \ y + x \ OR \ y = x + y$, so the total sum of all the elements never changes.
+
+**Observation 2**: Notice that during each operation, the number and position of the set bits don't change. For instance, let $x = 110_2$, $y = 101_2$. $x \ AND \ y = 100_2$, and $x \ OR \ y = 111_2$. Before and after the operation, there are two $1$ bits that are set, one $2$ bit that is set, and one $4$ bit that is set.
+
+**Observation 3**: Notice when performing an operation, $\sum_{i=1}^{n}{a_i^2}$ increases when $|x - y|$ increases. Thus, we want to keep performing operations such that $|x - y|$ increases. In fact, because $x \ AND \ y \leq \min(x, y)$ and $x \ OR \ y \geq \max(x, y)$, performing an operation will never decrease $|x - y|$.
+
```

Comment:
it'd be great if you could structure these as hints! (see https://github.com/cpinitiative/usaco-guide/pull/3430)